File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        // 도난당한 학생과 여벌이 있는 학생을 정렬
        Arrays.sort(lost);
        Arrays.sort(reserve);

        // 도난당했지만 여벌이 있는 학생들을 처리 (교집합 제거)
        List<Integer> lostList = Arrays.stream(lost)
                .filter(l -> !IntStream.of(reserve).anyMatch(r -> r == l))
                .boxed()
                .collect(Collectors.toList());
        List<Integer> reserveList = Arrays.stream(reserve)
                .filter(r -> !IntStream.of(lost).anyMatch(l -> l == r))
                .boxed()
                .collect(Collectors.toList());

        // 여벌 체육복을 빌려주기
        for (int r : reserveList) {
            if (lostList.contains(r - 1)) {
                lostList.remove(Integer.valueOf(r - 1));
            } else if (lostList.contains(r + 1)) {
                lostList.remove(Integer.valueOf(r + 1));
            }
        }

        // 전체 학생에서 빌리지 못한 학생 수를 뺀다
        return n - lostList.size();
    }
}

File: 프로그래머스/1/42862. 체육복/체육복.cpp
#include <algorithm>
#include <vector>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int maxStd =n;

    sort(lost.begin(), lost.end());
    sort(reserve.begin(), reserve.end());
    // 2 3 4
    // 2 3 4
    vector<int> rlost;
    
    for (int l : lost) {
        if (find(reserve.begin(), reserve.end(), l) != reserve.end()) {
            reserve.erase(find(reserve.begin(), reserve.end(), l));
        }else {
            rlost.push_back(l);
        }
    }


    for (int l : rlost) {
        if (find(reserve.begin(), reserve.end(), l-1) != reserve.end()) {
            reserve.erase(find(reserve.begin(), reserve.end(), l-1));
            continue;
        }

        if (find(reserve.begin(), reserve.end(), l+1) != reserve.end()) {
            reserve.erase(find(reserve.begin(),reserve.end(),l+1));
            continue;
        }
        maxStd--;
    }
    return maxStd;
}